src/vscode-dts/vscode.proposed.textEditorDragAndDrop.d.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/142990
+
+	export class TextEditorDataTransferItem {
+		asString(): Thenable<string>;
+		readonly value: any;
+		constructor(value: any);
+	}
+
+	export class TextEditorDataTransfer<T extends TextEditorDataTransferItem = TextEditorDataTransferItem> {
```

Comment:
This is copy pasted from the tree drag drop provider prototype. We should look into unifying these types 

/cc @alexr00